src/App.tsx

```diff
@@ -111,6 +112,10 @@ function App({ isEmbedded = false }: { isEmbedded?: boolean }) {
     });
   }, [machine?.id, sendToSourceService]);
 
+  useEffect(() => {
+    analytics()?.track('Opening XState Viz');
+  }, []);
```

Comment:
It's literally a thing that should happen once - I'd move this to the module scope. `useEffect` feels redundant here.

Reply:
Are we fine with this being called this early? The `.identify` call might not happen yet at this point in time. But perhaps MixPanel is smart enough to correlate those 2 things based on their own user ID?

Reply:
We should verify the order here. Rather than call `track('Opening XState Viz');` in the module scope, we may need to defer the call.

Reply:
@Andarist You're right, the `identify()` call was being made after the initial tracking call in the App's `useEffect` which explains why the id wasn't picked up in the payload of the tracking call.